src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp

```diff
@@ -31,27 +31,33 @@ class DiscoveryDataQueueInfo
 {
 public:
 
-    DiscoveryDataQueueInfo()
+    DiscoveryDataQueueInfo(
+            eprosima::fastrtps::rtps::CacheChange_t* cache_change,
+            eprosima::fastrtps::string_255 topic_name)
+        : cache_change_(cache_change)
+        , topic_name_(topic_name)
     {
     }
 
     ~DiscoveryDataQueueInfo()
     {
     }
 
-    eprosima::fastrtps::rtps::CacheChange_t* change();
-
-    eprosima::fastrtps::string_255 topic();
+    eprosima::fastrtps::rtps::CacheChange_t* cache_change()
```

Comment:
In line with the rest of the structures and the design:
```suggestion
    eprosima::fastrtps::rtps::CacheChange_t* change()
```